### ISSUE TITLE
Deck image build flakiness: sometimes spyglass doesn't load, rebuild image works

### ISSUE THREAD

**Issue by chaodaiG** [kubernetes/test-infra#26345]
**What happened**:

It had happened twice, that after prow bump, spyglass pages didn't load, and it worked after rebuilding deck image with the same revision.

Image didn't work: gcr.io/k8s-prow/deck:v20220516-bc723c0e51
Rebuilt: gcr.io/k8s-prow/deck:v20220517-bc723c0e51

**What you expected to happen**:

Deck image build is deterministic.

**How to reproduce it (as minimally and precisely as possible)**:

When it happened, click the retest button on prow https://prow.k8s.io?job=post-test-infra-push-prow and use the newly built image for prow.

**Please provide links to example occurrences, if any**:

ref: https://github.com/kubernetes/test-infra/issues/26344

**Anything else we need to know?**:

The symptom was the same when it happened last time.

/sig testing
/assign

**Comment by chaodaiG** [comment]
Quickly inspected the spyglass page when the error was happening, found out that there was nothing loaded in iframe:

<img width="952" alt="Screen Shot 2022-05-17 at 7 48 01 AM" src="https://user-images.githubusercontent.com/45011425/168846227-71ba5dd8-e8a5-48af-885d-f6b7f6ba3fca.png">


And after the image rebuilt, iframes were loaded:

<img width="952" alt="Screen Shot 2022-05-17 at 7 52 31 AM" src="https://user-images.githubusercontent.com/45011425/168846346-9839a4ff-ff50-4447-bd68-c445df792a89.png">

**Comment by chaodaiG** [comment]
Downloaded both images gcr.io/k8s-prow/deck:v20220516-bc723c0e51 and gcr.io/k8s-prow/deck:v20220517-bc723c0e51 locally, extracted them and comparing the differences. The first noticeable difference is `spyglass_bundle.min.js` which is compiled as part of image build process.

The size of the file is very different:
- 0516: 4581b
- 0517: 64273b

This is very likely related as the symptom is directly from spyglass. Since this file is generated by https://github.com/kubernetes/test-infra/blob/40939310cddb7abbb3aca3fda78152cec4f5b303/prow/cmd/deck/gather-static.sh during image building process, something might have happened there

**Comment by chaodaiG** [comment]
This appears mystery to me, starring at the source code of the typescript compilation and didn't see anything obviously wrong. This is probably not that easily reproducible which makes the debugging more difficult.

One way to solve this problem, is that we could source control all of these bundled js files, and use verify-codegen to guard against them in presubmit. The problem with this approach, is that it will be confusing to users when it happens in presubmit test, the result will suggest rerun `update-codegen` and when user runs it there will be no change, the only way to get around is retest the prow job.

**Comment by BenTheElder** [comment]
Worth noting that we're using *really* outdated typescript / node tooling. We should probably switch to something up to date, if there is a bug in any downstream tools we're _way_ behind on any patches.

**Comment by oliver-goetz** [comment]
https://github.com/kubernetes/test-infra/issues/25998 might be related to this. 
If flakiness was the cause here too, revert "Deck: Sort consistently" (https://github.com/kubernetes/test-infra/pull/26013) would have been not needed here.

**Comment by chaodaiG** [comment]
> #25998 might be related to this. If flakiness was the cause here too, revert "Deck: Sort consistently" (#26013) would have been not needed here.

that's correct, #25998 was exactly the last time we saw this, and reverting #26013 had been proven not necessary from last time as well

**Comment by oliver-goetz** [comment]
ahh there is already a "Revert: Revert..." Didn't see that 😅

**Comment by chaodaiG** [comment]
this happened again today :(

**Comment by timebertt** [comment]
This happened again with the bump from `v20220621-6ff60de3c2` to `v20220623-3ef0bc3b81`.
It might be worth holding or skipping https://github.com/kubernetes/test-infra/pull/26653 before it affects prow.k8s.io as well.

**Comment by BenTheElder** [comment]
xref: https://github.com/kubernetes/test-infra/issues/26655 looks like the latest instance

**Comment by chaodaiG** [comment]
This happened again after the landing of https://github.com/kubernetes/test-infra/pull/26902, and there were more logs available now.

Image building jobs:
- Bad one: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-test-infra-push-prow/1549398192859123712
- Good one: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-test-infra-push-prow/1549507684703670272

Suspicious line of logs:
```
_output/js/prow/cmd/deck/static/spyglass/lens.js → _output/js/prow/cmd/deck/static/spyglass/bundle.js...
_output/js/prow/cmd/deck/static/spyglass/spyglass.js → _output/js/prow/cmd/deck/static/spyglass/bundle.js...
```

Basically two processes writing to the same file, it works when they don't happen at the same time, otherwise we are in trouble, e.g. deck image becomes invalid.

Created #26902 fixing this issue.
